sdk/core/azure-core/azure/core/messaging.py

```diff
@@ -141,17 +142,13 @@ def from_dict(cls, event, **kwargs):
         if data and data_base64:
             raise ValueError("Invalid input. Only one of data and data_base64 must be present.")
 
-        try:
-            time = isodate.parse_datetime(event.get("time", None))
-        except AttributeError:
-            time = None
         return cls(
             id=event.get("id", None),
             source=event.get("source", None),
             type=event.get("type", None),
             specversion=event.get("specversion", None),
             data=data if data is not None else b64decode(data_base64),
-            time=time,
+            time=_convert_to_isoformat(event.get("time", None)),
```

Comment:
we cannot unfortunately use `​fromisoformat`​ even in 3.7 + - it does not support Zulu time (Z suffix) which is the format that is being used in storage queue, service bus queue